presto-main/src/main/java/com/facebook/presto/sql/relational/DeferredSymbolReferenceExpression.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.relational;
+
+import com.facebook.presto.spi.type.Type;
+
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class DeferredSymbolReferenceExpression
```

Comment:
If this class is introduced only to fit into `ExpressionEquivalence` mechanisms, why not add an exception for that case there, instead of unnecessarily implementing many visitors with `throw new UnsupportedOperationException()` or `return null`?
You could use `ExpressionExtractor` to strip `DeferredSymbolReferences` before passing them to `EqualityInference`.